patterns/1-initial/incubator-pipeline.md

```diff
@@ -0,0 +1,90 @@
+## Title
+
+Incubator Pipeline
+
+## Patlet
+
+A team maintains a shared code library that other internal engineers need to use. The other engineers want to make improvements or use their own code. Leveraging an incubator model, the shared library team can establish a lower bar to enter the incubator pipeline and a higher bar to exit the pipeline and become a top-level unit in the library.
+
+## Problem
+
+A team charged to maintain a shared library is responsible for the quality of the components in the library. If you offer them code from another team, they face the dilemma that the other team’s code might not meet their requirements, and it’s unclear if they’ll be able to support that code. If other teams can’t offer code, they may be reluctant to use the shared code, or they may be forced to use inadequate code from the central team, causing issues for their product.
+
+The team managing the shared code library needs a way to allow InnerSource contributions to their supported library of components but also to distinguish between contributions that are up to the standard, or that are on their way to the standard.
+
+## Story
+
+See https://www.linkedin.com/pulse/culture-behaviors-innersource-three-part-blog-series-3-gil-yehuda/ for the conceptual inspiration. (This may be replaced with a direct story from the dev team who implemented this pattern. Watch this space.)
+
+## Context
+
+We find the pattern to be useful when there’s a team at a company that is responsible for creating code assets that other engineering teams at the company must consume. For example:
+
+1. the team that creates the centrally approved library of UI components,
+1. the team that builds the build pipelines and installation scripts,
+1. the team that builds microservice template code, etc.
+
+These teams are held to meet certain standards (e.g., UI teams ensure all components comply with accessibility standards, can be themed and placed on the UI grid in a manner consistent with other components and company requirements; pipeline and install scripts might require certain boilerplate code for compliance logging or security controls, etc.).
+
+However, the shared code library team does not want engineers to “roll their own” solutions either. But the user-teams may have needs not met by the shared library team’s code. Since they are engineers, they’ll want to create or find their own solutions. Doing that threatens the shared library team and creates multiple solutions. Whereas that’s sometimes okay, in some cases companies want to use InnerSource to maintain consistency without stifling innovation.  
+
```

Comment:
@gyehuda I just had one other thought:
Is it essential for this pattern to work that the library in question is modular, so that contributions from an out-group that are under incubation can be kept "separate" from the rest of the code?

If so, then this would belong into the Context as it would help readers to understand "do I have this situation?".

Reply:
In the meantime we have this new plugin that talks about extensions/plugins to enable InnerSource. Could be worth linking to. https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/extensions-to-manage-contributions-at-scale.md